Parser/pegen/parse_string.c

```diff
@@ -12,20 +12,27 @@
 // file (like "_PyPegen_raise_syntax_error").
 
 static int
-warn_invalid_escape_sequence(Parser *p, unsigned char first_invalid_escape_char)
+warn_invalid_escape_sequence(Parser *p, unsigned char first_invalid_escape_char, Token *t)
 {
     PyObject *msg =
         PyUnicode_FromFormat("invalid escape sequence \\%c", first_invalid_escape_char);
     if (msg == NULL) {
         return -1;
     }
     if (PyErr_WarnExplicitObject(PyExc_DeprecationWarning, msg, p->tok->filename,
-                                 p->tok->lineno, NULL, NULL) < 0) {
+                                 t->lineno, NULL, NULL) < 0) {
         if (PyErr_ExceptionMatches(PyExc_DeprecationWarning)) {
             /* Replace the DeprecationWarning exception with a SyntaxError
                to get a more accurate error report */
             PyErr_Clear();
+
+            /* This is a hack, in order for the SyntaxError to point to the token t,
+               since _PyPegen_raise_error always uses p->tokens[p->fill - 1] for the
+               error location. */
+            Token *old_end = p->tokens[p->fill - 1];
+            p->tokens[p->fill - 1] = t;
```

Comment:
Instead, could we have a new optional `Token*` attribute on the parser that (optionally) points to a token where the syntax error happens? Then, in the raise function, we take that token as a preference over `p->tokens[p->fill - 1]`. In this way, we don't need to modify the token string, which initially feels suboptimal (I have not thought about other side effects this may had).

Reply:
You mean in the `Parser` struct?

Reply:
> You mean in the `Parser` struct?

Yup

Reply:
Maybe `p->known_err_token` or something similar.

Reply:
Good idea! On it.

Reply:
Done.